<launch>
    <arg name="usb" default="0"/>
    <arg name="reach_tolerance_start" default="1"/>
    <arg name="reach_tolerance_end" default="0.2"/>
    <arg name="ns_prefix" default="uav_"/>
    <arg name="id" default="1"/>
    <arg name="trajectory" default="false"/>
    <arg name="generator_mode" default="0"/>
    <arg name="save_test_data" default="false"/>
    <arg name="save_experiment_data" default="false"/>
    <arg name="look_ahead" default="0.4"/>
    <arg name="cruising_speed" default="1.0"/>

    <!-- Launch robot id=2 -->
    <include file="$(find px4_bringup)/launch/spawn_robot.launch" ns="$(arg ns_prefix)$(arg id)">
        <arg name="id" value="$(arg id)"/>
        <arg name="mode" value="custom"/>
        <arg name="fcu_url" value="serial:///dev/ttyUSB$(arg usb):921600"/>
    </include>

    <!-- Launch UAL server for drone id=2 -->
    <node pkg="uav_abstraction_layer" type="server" name="ual" output="screen" ns="$(arg ns_prefix)$(arg id)">
        <param name="uav_id" value="$(arg id)"/>
        <param name="pose_frame_id" value="map"/>
        <rosparam param="map_origin_geo">[37.5585339, -5.9310622, 7.89]</rosparam>
        <rosparam param="home_pose">[-32.11, 4.01, 0]</rosparam>        <!-- [x,y,z] -->
    </node>

    <!-- UAV Path Manager nodes -->
    <group ns="upat_follower">
        <group ns="$(arg ns_prefix)$(arg id)">
            <node pkg="upat_follower" type="ual_communication_node" name="ual_communication" output="screen" required="true">
                <param name="uav_id" value="$(arg id)"/>
                <param name="save_test_data" value="false"/>
                <param name="pub_rate" value="100"/>
                <param name="path" value="cubic"/>
                <param name="reach_tolerance_start" value="$(arg reach_tolerance_start)"/>
                <param name="reach_tolerance_end" value="$(arg reach_tolerance_end)"/>
                <param name="use_class" value="true"/>
                <param name="trajectory" value="$(arg trajectory)"/>
                <param name="generator_mode" value="$(arg generator_mode)"/>
                <param name="look_ahead" value="$(arg look_ahead)"/>
                <param name="cruising_speed" value="$(arg cruising_speed)"/>
            </node>
            <node pkg="upat_follower" type="visualization_node" name="visualization" required="true" output="screen">
                <param name="uav_id" value="$(arg id)"/>
                <param name="robot_model" value="iris"/>
                <param name="pub_rate" value="100"/>
                <param name="trajectory" value="$(arg trajectory)"/>
                <param name="generator_mode" value="$(arg generator_mode)"/>
                <param name="save_experiment_data" value="$(arg save_experiment_data)"/>
            </node>
        </group>
    </group>

</launch>